app/main.py

```diff
@@ -1 +1,23 @@
-# write your code here
+from __future__ import annotations
+from app.cafe import Cafe
+from app.errors import (VaccineError, NotWearingMaskError)
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    try:
+        sorted_friends = sorted(friends,
+                                key=lambda x: x["vaccine"]["expiration_date"])
```

Comment:
Because if one of friends with bigger date of vaccine and no mask goes earlier in the list of friends, that I check, I get a NotWearingMaskError and can miss someone in the list with expired vaccine date, whose position in the list is further.
So I desided to sort dictionaries by vaccine expiration date to get the correct exception.